src/binwalk/magic/firmware

```diff
@@ -271,7 +271,7 @@
 #
 # Motorola S-Records, from Gerd Truschinski <gt@freebsd.first.gmd.de>
 # Improved by Martin Sundhaug <martinsundhaug@gmail.com>
-0   regex       (S[0-35-9]([0-9A-F]{4})([0-9A-F]{2})+\n)+       Motorola S-Record{many}; binary data in text format, record type:
```

Comment:
Huh, didn't know about the block-reading (not sure if it's documented). The test I did at the time was case 3 with a small file (less than 1 MB), thinking it also covered case 2. Solving case 1 probably requires a plugin to solve and some heuristics (since S-records can come without the header) with the presumption that either based on "redefines" or the presumption of record-order. Could case 2, like case 1, be solved using a plugin or is it better to improve how {many}-works to solve case 3?